pinc/DPDatabase.inc

```diff
@@ -59,12 +59,32 @@ final class DPDatabase
         return self::$_connection;
     }
 
-    static public function log_error()
+    static public function escape($value)
+    {
+        return mysqli_real_escape_string(self::$_connection, $value);
+    }
+
+    static public function query($sql, $except_on_failure=TRUE)
```

Comment:
```suggestion
    static public function query($sql, $exception_on_failure=TRUE)
```
... because "except" could be read as "Do something, *except* when there's a failure."

Or maybe change it to `$throw_on_failure`.

Reply:
I like `$throw_on_failure` better, good idea.